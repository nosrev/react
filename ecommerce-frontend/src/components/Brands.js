import React from 'react';
import brand1 from '../img/brand1.png';
import brand2 from '../img/brand2.png';
import brand3 from '../img/brand3.png';
import brand4 from '../img/brand4.png';
import { ReactComponent as Left } from '../svg/left.svg';
import { ReactComponent as Right } from '../svg/right.svg';
import Slider from "react-slick";
import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";

const Brands = () => {
    function PrevArrow(props) {
      const { className, style, onClick } = props;
      return (
        <div
          className={className}
          style={style}
          onClick={onClick}
        >
            <Left className="icon w-8 left" />
        </div>
      );
    }

    function NextArrow(props) {
      const { className, style, onClick } = props;
      return (
        <div
          className={className}
          style={style}
          onClick={onClick}
        >
            <Right className="icon w-8 right" />
        </div>
      );
    }

    const settings = {
        slidesToShow: 4,
        nextArrow: <NextArrow />,
        prevArrow: <PrevArrow />,
        draggable: true,
        responsive: [
            {
              breakpoint: 680,
              settings: {
                slidesToShow: 3,
              }
            },
            {
              breakpoint: 440,
              settings: {
                slidesToShow: 2,
              }
            },
            {
              breakpoint: 380,
              settings: {
                slidesToShow: 1,
              }
            },
        ]
    };

    return (
        <div className="brands-wrapper">
            <Slider {...settings} className="brands">
                <img src={brand1} alt="brand1" />
                <img src={brand2} alt="brand2" />
                <img src={brand3} alt="brand3" />
                <img src={brand4} alt="brand4" />
                <img src={brand2} alt="brand2" />
                <img src={brand1} alt="brand1" />
                <img src={brand3} alt="brand3" />
            </Slider>
        </div>
    )
}

export default Brands;
