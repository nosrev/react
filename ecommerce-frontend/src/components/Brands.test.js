import renderer from 'react-test-renderer';
import Brands from './Brands';
import { MemoryRouter as Router } from 'react-router-dom';

it('render Brands', () => {
  const component = renderer.create(
    <Brands />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
