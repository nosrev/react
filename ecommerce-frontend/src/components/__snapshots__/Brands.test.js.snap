// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Brands 1`] = `
<div
  className="brands-wrapper"
>
  <div
    className="slick-slider brands slick-initialized"
    dir="ltr"
  >
    <div
      className="slick-arrow slick-prev"
      onClick={[Function]}
      style={
        Object {
          "display": "block",
        }
      }
    >
      <svg
        className="icon w-8 left"
      >
        left.svg
      </svg>
    </div>
    <div
      className="slick-list"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={null}
      onMouseMove={null}
      onMouseUp={[Function]}
      onTouchCancel={null}
      onTouchEnd={[Function]}
      onTouchMove={null}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <div
        className="slick-track"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "WebkitTransform": "translate3d(0px, 0px, 0px)",
            "WebkitTransition": "",
            "msTransform": "translateX(0px)",
            "opacity": 1,
            "transform": "translate3d(0px, 0px, 0px)",
            "transition": "",
          }
        }
      >
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={-4}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand4"
              src="brand4.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={-3}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand2"
              src="brand2.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={-2}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand1"
              src="brand1.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={-1}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand3"
              src="brand3.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active slick-current"
          data-index={0}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand1"
              src="brand1.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={1}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand2"
              src="brand2.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={2}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand3"
              src="brand3.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={false}
          className="slick-slide slick-active"
          data-index={3}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand4"
              src="brand4.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={4}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand2"
              src="brand2.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={5}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand1"
              src="brand1.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide"
          data-index={6}
          onClick={[Function]}
          style={
            Object {
              "outline": "none",
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand3"
              src="brand3.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={7}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand1"
              src="brand1.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={8}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand2"
              src="brand2.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={9}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand3"
              src="brand3.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={10}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand4"
              src="brand4.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={11}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand2"
              src="brand2.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={12}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand1"
              src="brand1.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
        <div
          aria-hidden={true}
          className="slick-slide slick-cloned"
          data-index={13}
          onClick={[Function]}
          style={
            Object {
              "width": 0,
            }
          }
          tabIndex="-1"
        >
          <div>
            <img
              alt="brand3"
              src="brand3.png"
              style={
                Object {
                  "display": "inline-block",
                  "width": "100%",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="slick-arrow slick-next"
      onClick={[Function]}
      style={
        Object {
          "display": "block",
        }
      }
    >
      <svg
        className="icon w-8 right"
      >
        right.svg
      </svg>
    </div>
  </div>
</div>
`;
